### PR TITLE
feat(link): add experimental link component

### PR DESCRIPTION
**Pull request by chrsfshr**
Adds experimental link component.

Staging Link: http://carbon-dev-experimental-grumpy-mongoose.mybluemix.net/demo/link

Issue Reference: #1108

### CODE REVIEW

**Comment by jeanservaas** [comment]
### Link feedback
<br>
1. Font weight (both :enabled and :disabled)
<br><br>

- **You’ve got it at font-weight:600 — that’s the semi-bold; it needs to be Plex-Sans Regular, which is font-weight: 400** 
<br>
2. Underline Offset
<br><br>

-  	**Correction: Offset underline 3px from copy** _(See screen grab, Link_underline offset)_



link_underline_offset

**Comment by chrsfshr** [comment]
@jeanservaas Updated component based on feedback, let me know if there are any more changes to be made. Thanks!

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
hey @chrsfshr the link component looks great!! 🎉 

looks like some of your accordion updates got added in to this PR by mistake, can you remove before we merge

File: src/components/link/_link.scss
Comment (alisonjoseph) [inline src/components/link/_link.scss]: should this be replaced with a token?

Comment (alisonjoseph) [reply src/components/link/_link.scss]: look in `_theme.scss` at `@mixin theme--experimental() {` for the tokens 😄 

so this would be ` $brand-01`

File: src/components/link/_link.scss
Comment (alisonjoseph) [inline src/components/link/_link.scss]: might be good to add a link token here` $link-visited` maybe?

File: src/components/link/_link.scss
Comment (alisonjoseph) [inline src/components/link/_link.scss]: I added a token to the text input PR `$text-disabled` you could use

Comment (chrsfshr) [reply src/components/link/_link.scss]: Was going to wait until you merged your `text-input` PR to change the token, didn't want to override what you have.

**Comment by alisonjoseph** [comment]
thanks @chrsfshr! I went ahead and updated the PR to add the variable I had in the same way.

**Review by alisonjoseph — APPROVED** [review]
LGTM 

*added a style for visited:active state.

**Comment by alisonjoseph** [comment]
@chrsfshr can you get my latest changes and publish an updated staging link and then assign to Jeannie to review?

**Comment by chrsfshr** [comment]
Thanks @alisonjoseph! 

@jeanservaas Can you take one last past at the updated staging link to confirm design sign-off. Thanks!

**Comment by alisonjoseph** [comment]
Hey @chrsfshr looks like we can only have one "Random" url posted at a time so this staging link is showing the accordion updates and not link (same thing happened to me when I was trying to post two different links). 

If you change the name field in the `manifest.yml` file and then push I think it will generate a different url. 🤞

**Comment by chrsfshr** [comment]
Staging link updated!

**Comment by jeanservaas** [comment]
Link is GOOD, thumbs UP merge

**Review by joshblack — APPROVED** [review]

**Comment by jeanservaas** [comment]
@chrsfshr Here's button.

Note: Carbon has both a Primary Danger button and a danger button... we have only one danger button... I was going to ask Anna about this, if we need to add a second danger button we can, but in the meantime I wanted to get this to you.

[Button.zip](https://github.com/IBM/carbon-components/files/2407000/Button.zip)

**Pull request merged**
